### DISCUSSION TITLE
[FEATURE] Show album view when click on cluster with more than 10 photos on map

### DISCUSSION THREAD

**Discussion by osresearch** [immich-app/immich#3218] [General]
### The bug

![Screenshot_2023-07-11_11-39-11](https://github.com/immich-app/immich/assets/3068843/b443124e-8456-4f86-ab20-a1616f56cc22)

When there are only a few photos in a map cluster, clicking on them brings up a pie menu to choose from the photos, but if there are many (more than 10?) it seems to choose one from the cluster and displays the lightbox for that one photo instead.

### The OS that Immich Server is running on

Ubuntu 20.04

### Version of Immich Server

v1.66.1

### Version of Immich Mobile App

none

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
the default
```

### Your .env content

```Shell
the default
```

### Reproduction steps

```bash
1. Upload lots of photos from one location
2. Go to the map view at a zoomed out level so that more than 10 are clustered together
3. Click on the cluster
4. Receive a random photo
...
```

**Comment by lifeart** [comment]
It seems we should not fallback to `0` here: https://github.com/immich-app/immich/blob/main/web/src/lib/components/shared-components/leaflet/marker-cluster/asset-marker-cluster.svelte#L88

If items > 10, first one is chosen: https://github.com/immich-app/immich/blob/main/web/src/lib/components/shared-components/leaflet/marker-cluster/asset-marker-cluster.svelte#L58

**Comment by osresearch** [comment]
![IMG_4795](https://github.com/immich-app/immich/assets/3068843/5557cdcf-3969-4831-92f5-a3eb24d262bd)

If the cluster is too large, a new feature that would be really nice is to open an album view of the photos in the cluster.  Some other systems do this with their heatmap view.

**Reply by enteryournamehere** [reply]
For some more inspiration:
Nextcloud Memories has a live-updating album view next to the map, showing all images currently in view on the map. Clicking on a cluster either zooms in a little bit, or if you're zoomed in far enough, it opens the targeted image in full screen, and you can move left & right to view the rest of the previously mentioned 'album' (everything currently in view on the map). Desktop screenshot:
![nextcloud_memories_desktop](https://github.com/immich-app/immich/assets/11255568/e66eeb11-60d9-4711-9284-0e9ef835979f)
Mobile screenshot:
<img width="300" src="https://github.com/immich-app/immich/assets/11255568/52d3651b-627e-4e26-814e-0b34cf49f1fe"/>

Synology Photos lets you tap a cluster and it'll open just that cluster in an album view. (The browser version doesn't seem to have a map.)
<img width="300" src="https://github.com/immich-app/immich/assets/11255568/ed38ec38-f18f-4e52-97cf-606f9c3dac75"/> <img width="300" src="https://github.com/immich-app/immich/assets/11255568/8828988f-b215-4c8f-9bc7-53d9a610f874"/>

**Reply by turkishgetup** [reply]
> For some more inspiration: Nextcloud Memories has a live-updating album view next to the map, showing all images currently in view on the map. Clicking on a cluster either zooms in a little bit, or if you're zoomed in far enough, it opens the targeted image in full screen, and you can move left & right to view the rest of the previously mentioned 'album' (everything currently in view on the map). 

I came to this discussion thinking I'd open my own feature request: it would be so nice to have the map and lightbox view simultaneously and then the map live-updating. 
Usecase: I open a single photo, e.g. by discovering it as a memory. Then I see on the map that I have more photos from the same geographic region, that I then get to explore easily via the map.

**Comment by LaurentAtJoes** [comment]
not sure what the current status is for this approved feature request to "show album" (let's say: grid view instead of 1 picture fullscreen), but it sure would be very nice.

**Comment by Spacey-Rat** [comment]
Hi!
same - when I click in a browser on my computer (not mobile) on a location with eg 300 pictures I'm ending up directly in the picture view and have to scroll through all the 300 pictures. A grid view (album view) with only the selected location would be great in between the final picture view and the location view. Absolut great project - highly appreciated!! Thx..

**Comment by onrust** [comment]
I'd like to add my vote to this feature request.

**Reply by enteryournamehere** [reply]
I believe this discussion has been closed as a duplicate in favour of #4519.

**Comment by Fifou2202** [comment]
+1 – I strongly support this feature.

I often use the map to find photos taken at very specific locations using GPS metadata. Right now, clicking on a location only shows one photo, and I have to swipe through manually.

It would be incredibly useful to view all photos in a gallery/grid mode, especially to bulk delete pictures from precise spots (e.g., test shots, duplicates, random locations). Much more accurate than filtering by city name. Thanks!
